app/models/test.rb

```diff
@@ -1,8 +1,24 @@
 class Test < ApplicationRecord
+  belongs_to :category
+  belongs_to :author, class_name: 'User', inverse_of: 'created_tests'
+
+  has_many :questions, dependent: :destroy
+  has_many :results, dependent: :destroy
+  has_many :users, through: :results
+
+  validates :title, presence: true
+  validates :title, presence: true, uniqueness: { scope: :level }
```

Comment:
[в документации](https://guides.rubyonrails.org/active_record_validations.html#uniqueness) к хелперу `uniqueness` сказано следующее:
> It does not create a uniqueness constraint in the database, so it may happen that two different database connections create two records with the same value for a column that you intend to be unique. To avoid that, you must create a unique index on that column in your database.

другими словами, чтобы эта была настоящая уникальность (не только на уровне приложения, но и на уровне базы), нужно в базе создать уникальный индекс

про правильную миграцию можно посмотреть к примеру на [СО](https://stackoverflow.com/questions/32927262/create-migration-with-uniqueness-to-a-scope)

примерно таким образом:

```ruby
class AddIndexToTests < ActiveRecord::Migration[5.2]
  def change
    add_index :tests, [:title, :level], unique: true
  end
end
```

Reply:
done